Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -806,8 +813,18 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
         else {
             DFG_ASSERT(m_graph, node, node->binaryUseKind() == UntypedUse);
             clobberWorld();
-            // FIXME: do we really need SpecString here for ValueSub? It seems like we only need it for ValueAdd.
-            setTypeForNode(node, SpecString | SpecBytecodeNumber | SpecBigInt);
+            if (node->op() == ValueAdd)
+                setTypeForNode(node, SpecString | SpecBytecodeNumber | SpecBigInt);
```

Comment:
Could you still try to limit the result type if both `child1` and `child2` are numbers?

Reply:
That pattern is handled in ArithAdd / ArithSub. Here, we are handling the cases that ArithAdd / ArithSub missed to handle well, like, `something - 20` for example.